src/tribler/core/components/gigachannel/community/tests/test_gigachannel_community.py

```diff
@@ -12,31 +14,54 @@
     ChannelsPeersMapping,
     GigaChannelCommunity,
     NoChannelSourcesException,
+    happy_eyeballs_delay
 )
 from tribler.core.components.gigachannel.community.settings import ChantSettings
 from tribler.core.components.metadata_store.db.store import MetadataStore
 from tribler.core.components.metadata_store.remote_query_community.remote_query_community import EvaSelectRequest, \
-    SelectRequest
+    SelectRequest, RemoteSelectPayload, RemoteSelectPayloadEva, SelectResponsePayload
 from tribler.core.components.metadata_store.remote_query_community.settings import RemoteQueryCommunitySettings
 from tribler.core.components.metadata_store.utils import RequestTimeoutException
 from tribler.core.utilities.notifier import Notifier
 from tribler.core.utilities.path_util import Path
 from tribler.core.utilities.utilities import random_infohash
 
-EMPTY_BLOB = b""
-
 # pylint:disable=protected-access
 
+EMPTY_BLOB = b""
+U_CHANNEL = "ubuntu channel"
+U_TORRENT = "ubuntu torrent"
+CHANNEL_ID = 123
 BASE_PATH = 'tribler.core.components.metadata_store.remote_query_community.remote_query_community'
+ID1 = 0
+ID2 = 1
+ID3 = 2
+
+
+@dataclass
+class ChannelKey(Mapping):
+    channel_pk: bytes
+    origin_id: int
+
+    # The following methods allow for use as a Mapping (i.e., the "**key"-syntax)
+    def __iter__(self):
+        return iter(asdict(self))
+
+    def __getitem__(self, item):
+        return getattr(self, item)
+
+    def __len__(self):
+        return len(fields(self))
```

Comment:
It looks like it is possible to use a [typed `NamedTuple`](https://docs.python.org/3/library/typing.html#typing.NamedTuple) here instead of dataclass.
```suggestion
class ChannelKey(NamedTuple):
    channel_pk: bytes
    origin_id: int
```

Then the iteration of items will be available out of the box. With this change, `mapping.add(peer, *key.values())` becomes `mapping.add(peer, *key)`, and `mds.get_entries(**key)` becomes `mds.get_entries(**key._asdict())`. Despite underscore, `_asdict` is an [official public method](https://docs.python.org/3/library/collections.html#collections.somenamedtuple._asdict) for `namedtuple`.

Reply:
Thanks for the suggestion: that was also my initial approach. The reason that I did not use `NamedTuple`, in the end, is that the context menu in de PyCharm IDE does not list `_asdict`:

![contextmenu](https://github.com/Tribler/tribler/assets/3630389/3b5cef47-4917-40cc-8b87-9dcd825b26c5)

If you absolutely insist on using `NamedTuple`, I'm also fine with using it though.